primitives/src/traits/dispute_api.rs

```diff
@@ -51,4 +51,62 @@ pub trait DisputeApi {
         market_id: &Self::MarketId,
         market: &Market<Self::AccountId, Self::BlockNumber, Self::Moment>,
     ) -> Result<Option<OutcomeReport>, DispatchError>;
+
+    /// Query the future resolution block of a disputed market.
+    /// Fails if the market dispute mechanism does not belong to the actual dispute mechanism.
+    ///
+    /// # Returns
+    ///
+    /// Returns the future resolution block if available, otherwise `None`.
+    fn get_auto_resolve(
```

Comment:
I wonder why this is part of this interface. It seems to only be used when clearing auto-resolves, so I think we should just cache the auto-resolve in `prediction-markets` when we call `add_auto_resolve`.

Reply:
I somewhat agree to @Chralt98 's idea how ever it feels that by name `on_resolution()` and `get_auto_resolve()` both should go to `DisputeResolutionApi`. Or better we can merge both these traits and rename it to `DisputeMechanismApi` ?

Reply:
should this take a single dispute as input to the function ? Same question for `is_fail` ?

Reply:
> My aim for this PR is, that the MDM has full control over its resolution. This led me to the design decision to store the auto-resolve info inside the MDM.

Where you store the info has nothing to do with control. The MDM fully controls the resolution. Whether or not the interface of the MDM provides a way of querying the time of resolution or `prediction-markets` caches that value makes no difference to the amount of control that the MDM can exercise.

> Only the MDM should provide the information, when the resolution happens.

If the MDM provides the information of when the resolution happens, then `prediction-markets` should be allowed to cache that information. Caching when the resolution happens in `prediction-markets` actually makes it unnecessary for the MDM to expose that info, making `get_auto_resolve` unnecessary.

I'll think about this some more, but I hold no strong convictions in this case. I'm trying to make this interface a little simpler, but this comes at the price of some extra storage.

Reply:
>  how ever it feels that by name on_resolution() and get_auto_resolve() both should go to DisputeResolutionApi. Or better we can merge both these traits and rename it to DisputeMechanismApi ?

Yeah, you might think this way, because of the naming here. Both APIs have different purposes. `DisputeApi` is to be called from `prediction-markets` to the MDM. `DisputeResolutionApi` is to be called from an MDM to the `prediction-markets` pallet.

Do you have a better name for `DisputeResolutionApi`? The reason for choosing this was that an MDM is responsible now for its resolution.

Reply:
> Where you store the info has nothing to do with control. The MDM fully controls the resolution. Whether or not the interface of the MDM provides a way of querying the time of resolution or prediction-markets caches that value makes no difference to the amount of control that the MDM can exercise.

True. Say for example we have simple-disputes. Its resolution time is based on the last dispute, a simple calculation. There is no need to store additional information inside the storage, when we can calculate it from existing storage (which sits inside the pm-pallet (Disputes)). It just feels more intuitive to store that calculation, which is specific to simple-disputes, inside the simple-disputes pallet.

>  If the MDM provides the information of when the resolution happens, then prediction-markets should be allowed to cache that information. Caching when the resolution happens in prediction-markets actually makes it unnecessary for the MDM to expose that info, making get_auto_resolve unnecessary.

I see your point. Both solutions make sense. It's just, that I am in favour of not having more storage in this situation. I mean `get_auto_resolve` seems to be a useful optionality too.